os/arch/arm/src/amebasmart/amebasmart_smp.c

```diff
@@ -201,15 +168,30 @@ void smp_init(void)
 #if ( CONFIG_SMP_NCPUS > 1 )
 	for (xCoreID = 0; xCoreID < CONFIG_SMP_NCPUS; xCoreID++) {
 		if (xCoreID == up_cpu_index()) {
-			// pmu_set_secondary_cpu_state(xCoreID, CPU1_RUNNING);
+#ifdef CONFIG_CPU_HOTPLUG
+			up_set_secondary_cpu_state(xCoreID, CPU_RUNNING);
+#endif
 			continue;
 		}
-
 		err = psci_cpu_on(xCoreID, (unsigned long)__cpu1_start);
-		if (err < 0) {
-			printf("CPU%d: failed to boot: %d\n", (int)xCoreID, (int)err);
-		}
+		/* If we failed to boot secondary core here, it will be very likely
+		issue is coming from ATF/kernel flow, and that operation is irreversible
+		(ie. There is no way we can restore the secondary core to come
+		back to Tizen Lite, thus we should trigger a crash here to check)
+		*/
+		DEBUGASSERT(err >= 0);
+	}
+#endif
+	/* We should check whether it is a warm boot here
+	   If yes, we should send SGI1
+	 */
+#ifdef CONFIG_CPU_HOTPLUG
+	/* In our case, CONFIG_SMP_NCPUS is always 2, thus
+	xCoreID will be identified as the last core we have
+	from above loop, so we will use xCoreID here
+	*/
+	if (up_get_secondary_cpu_state(xCoreID) == CPU_WAKE_FROM_SLEEP) {
```

Comment:
When we come out of the for loop, the value of xCoreID will be 2. 
So, it will never enter this if condition. Please set core id to 1.

Reply:
Thank you for identifying this! Sorry for the mistake, I have fixed it.